turtle_graphics/turtle.js

```diff
@@ -0,0 +1,130 @@
+class Turtle {
+  constructor(initialX, initialY) {
+    this.x = initialX;
+    this.y = initialY;
+    this.path = [[this.x, this.y]];
+    this.direction = "x";
+    this.maxY = -Infinity;
+    this.minY = Infinity;
+    this.maxX = -Infinity;
+    this.minX = Infinity;
+  }
+
+  forward(steps) {
+    for (let i = 0; i < steps; i++) {
+      if (this.direction == "x") {
+        this.x++;
+      } else if (this.direction === "-y") {
+          this.y--;
+      } else if (this.direction === "-x") {
+        this.x--;
+      } else {
+        this.y++;
+      }
+      this.path.push([this.x, this.y]);
+    }
+    return this;
+  }
+
+  right() {
+    switch (this.direction) {
```

Comment:
I can see why you wouldn't like the `switch` syntax, but it does take some mental burden off of you once you get used to the "shape" of a switch (you no longer have to do the boolean comparison in your head as you traverse each `else if` branch). You can make this more compact (given the shortness of each branch) with something like:
```javascript
switch (this.direction) {
  case  "x": this.direction = "-y"; break;
  case  "y": this.direction =  "x"; break;
  case "-x": this.direction =  "y"; break;
  case "-y": this.direction = "-x"; break;
}
```
Given the symmetry of the branches, I think this use of inline statements is quite acceptable